openapi3/schema.go

```diff
@@ -956,17 +960,22 @@ func (schema *Schema) visitSetOperations(settings *schemaValidationSettings, val
 
 			return e
 		}
+
+		_ = v[matchedOneOfIdx].Value.visitJSON(settings, value)
 	}
 
 	if v := schema.AnyOf; len(v) > 0 {
 		ok := false
-		for _, item := range v {
+		matchedAnyOfIdx := 0
+		for idx, item := range v {
 			v := item.Value
 			if v == nil {
 				return foundUnresolvedRef(item.Ref)
 			}
-			if err := v.visitJSON(settings, value); err == nil {
+			tempValue := deepcopy.Copy(value)
```

Comment:
Move this outside the loop.

Reply:
Done.

Reply:
just fyi @nic-6443 @fenollp (ran into this PR while debugging)

Mutating `tempValue` across schemas listed under an `allOf` or `anyOf` brings about an issue in case two schemas declare a property with the same name but say a different default value, validation rules, etc.